js/src/setup-mc/setup-stepper/setup-accounts/index.js

```diff
@@ -23,23 +23,29 @@ import Faqs from './faqs';
 
 const SetupAccounts = ( props ) => {
 	const { onContinue = () => {} } = props;
-	const { jetpack } = useJetpackAccount();
-	const { google, scope } = useGoogleAccount();
-	const { googleMCAccount } = useGoogleMCAccount();
+	const {
+		jetpack,
+		hasFinishedResolution: hasFinishedResolutionJetpack,
+	} = useJetpackAccount();
+	const {
+		hasFinishedResolution: hasFinishedResolutionGoogle,
+	} = useGoogleAccount();
+	const {
+		googleMCAccount,
+		hasFinishedResolution: hasFinishedResolutionGoogleMC,
+	} = useGoogleMCAccount();
 
 	if (
-		! jetpack ||
-		( jetpack.active === 'yes' &&
-			( ! google || ( google.active === 'yes' && ! googleMCAccount ) ) )
+		! (
+			hasFinishedResolutionJetpack &&
+			hasFinishedResolutionGoogle &&
+			hasFinishedResolutionGoogleMC
+		)
```

Comment:
❓
Sometimes the `hasFinishedResolution` might be not so suitable to be used on the top layer. When the connection state of GMC changes and calls `invalidateResolution( 'getGoogleMCAccount', [] )` to updates data, it causes the whole page to flicker instantly. For example, at around the 6th second of this video, a flash occurs after GMC account is created:

https://user-images.githubusercontent.com/17420811/143012069-17cc604b-baac-4256-9f65-26321284d8d6.mp4

Reply:
Yeah, on hindsight, the impact is bigger than I thought. The change was done in [`090ecb9` (#1094)](https://github.com/woocommerce/google-listings-and-ads/pull/1094/commits/090ecb9f00485b0decc1a9633a9401f63341738d). I have now reverted the commit, done in [`64d110c` (#1094)](https://github.com/woocommerce/google-listings-and-ads/pull/1094/commits/64d110c492295fb8572c27db1a7a5f0c9dc15770). I have also added commit [`acaed2c` (#1094)](https://github.com/woocommerce/google-listings-and-ads/pull/1094/commits/acaed2cd2335be3a1af6dc63e61e372345947622), so that when we have a GMC account connection query after state change (e.g. we query for GMC account connection info after successfully connected a GMC account), it will display a `SpinnerCard`. All these should make it behave similar to existing production behavior.

I can work on the `hasFinishedResolution` enhancement changes later in a separate PR outside of the UX improvement project.

Reply:
Thanks for the adjustments! But now it will always get stuck on loading UI if users didn't provide GMC required permission at the first time. I think it's caused by this logic.

https://github.com/woocommerce/google-listings-and-ads/blob/acaed2cd2335be3a1af6dc63e61e372345947622/js/src/hooks/useGoogleMCAccount.js#L22-L28

Reply:
@eason9487 , thanks for the catch, again! 🙏 I have fixed this in [`f141a3a` (#1094)](https://github.com/woocommerce/google-listings-and-ads/pull/1094/commits/f141a3a35592b7c57352ef818d1963bad3def11d). (I should really stop doing too many things - even if they are minor improvements - in one PR. 😂 )

Reply:
> I should really stop doing too many things - even if they are minor improvements - in one PR. 😂

hahaha 🤣

Reply:
@eason9487 , I added 3 new commits https://github.com/woocommerce/google-listings-and-ads/pull/1094/files/f141a3a35592b7c57352ef818d1963bad3def11d..54366a7e56320d667f86a4e80b36e43691667aea to fix the loading state and disabled card issue. 🙏

Reply:
> it will always get stuck on loading UI if users didn't provide GMC required permission at the first time

I just realized that it can be fixed by doing an additional check on `scope.gmcRequired`, which I have just done in [`fc6d6d0` (#1094)](https://github.com/woocommerce/google-listings-and-ads/pull/1094/commits/fc6d6d023c25e71c4386a37cae38cbfe6b7be604). 😅

Reply:
Thanks! The transition of account states becomes much smoother. 🙌 

![Kapture 2021-11-24 at 17 57 53](https://user-images.githubusercontent.com/17420811/143216431-425114e0-0093-42d3-b201-1dc57831d470.gif)